### ISSUE TITLE
2C-2018-R

### ISSUE THREAD

**Issue by matsdah** [PI-ITBA/2023_02#73]
Buenas tardes, como va? Tenia una duda en el `l->tail = l->tail->tail;` luego de crear la lista aux. ¿Por que `l->tail` pasa a ser `l->tail->tail` y no `l->tail->elem`? Muchas gracias.

Consigna: Escribir una función recursiva que dada una lista de enteros que debería estar ordenada en forma
ascendente, elimine de la misma los elementos que no cumplan con ese orden.

Solucion de la catedra:
```c
void order(TList l){
  if( l==NULL || l->tail == NULL){
    return;
  }
  if(l->elem >= l->tail->elem){
    TList aux = l->tail;
    l->tail = l->tail->tail;
    free(aux);
    order(l);
  } else        // Sin el else igual funciona pero seria ineficiente
       order(l->tail);
}
```

**Comment by ImNotGone** [comment]
Si tenias lo siguiente

Direccion | elem | tail
-|-|-
0xA | 2 | 0xB
0xB | 1 | 0xC
0xC | 3 | NULL

Al estar parado en el 2, se cumple que el actual es mayor o igual al siguiente, por lo que elimina el 1.
Para hacer esto se guarda el nodo con el 1 en aux, actualiza el tail de 2 para q apunte al nodo que contiene el 3 y libera aux.

Por lo queda lo siguiente:
Direccion | elem | tail
-|-|-
0xA | 2 | **0xC**
0xC | 3 | NULL

**Comment by ImNotGone** [comment]
No estan en examen ahora igual?

**Comment by matsdah** [comment]
ando estudiando para el recu jeje

**Comment by ImNotGone** [comment]
Jajsjsjs mas allá de todo, recorda que elem es de tipo entero en este caso y tail es un puntero. Igualar tail a elem es un error conceptual. tail guarda siempre el valor de la direccion del siguiente nodo. Como nos estamos salteando el que tenemos adelante en este caso, usamos tail->tail para obtener el siguiente del siguiente. Nose si te termino de cerrar la idea

**Comment by matsdah** [comment]
Claro, me costaba entender que representaba el "tail del tail", es equivalente a decir el nodo "siguiente del siguiente" como decis vos. Muchas gracias!
